syncer/syncer.go

```diff
@@ -1714,6 +1730,13 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext) e
 		}
 	})
 
+	// NOTE: this DDL may be the first event for the table (no DML replicated before), so we try to save the table point (before this event) for it, then we can achieve:
+	// - can rollback tracked table info based on this checkpoint.
+	for db, tables := range sourceTbls {
+		for table := range tables {
+			s.saveTablePoint(db, table, lastLocation)
```

Comment:
we only try to rollback the tracked table info if it has checkpoint now, see https://github.com/pingcap/dm/blob/604de8d89a78d153cecdd3b0ea3543ab30a20503/syncer/checkpoint.go#L550.